refactor: SignUp 관련 코드 정리 및 데이터 구조 변경

FeelsBotMan opened the pull request:
- 선택된 포지션을 숫자 ID로 변경하여 SignUp 데이터에 포함

FeelsBotMan committed:
refactor: SignUp 관련 코드 정리 및 데이터 구조 변경

- SignUpInputs 인터페이스 이름 변경
- User.model에서 SignUpInputs를 가져와 사용하도록 수정
- 선택된 포지션을 숫자 ID로 변경하여 SignUp 데이터에 포함
- SignUpPage에서 포지션 버튼의 키를 ID로 변경

asdfbn12345 commented on src/pages/SignUpPage.tsx:
```diff
@@ -45,16 +42,16 @@ function SignUpPage() {
     alert: false,
   });
-  const [selectedPosition, setSelectedPostion] = useState<string | null>(null);
+  const [selectedPosition, setSelectedPostion] = useState<number | null>(null);
```
이름이 selectedPositionId면 number타입인게 유추하기 좋을 것 같습니다.

FeelsBotMan replied:
```ts
const [selectedPosition, setSelectedPostion] = useState<number>(1); 
```
으로 수정했습니다.

FeelsBotMan commented on src/pages/SignUpPage.tsx:
```diff
@@ -45,16 +42,16 @@ function SignUpPage() {
-    { value: "front-end", label: "Front-End" },
-    { value: "back-end", label: "Back-End" },
-    { value: "full-stack-developer", label: "Full-Stack-Developer" },
-    { value: "designer", label: "UI/UX Designer" },
+    { id: 1, value: "front-end", label: "Front-End" },
+    { id: 2, value: "back-end", label: "Back-End" },
+    { id: 3, value: "full-stack-developer", label: "Full-Stack-Developer" },
+    { id: 4, value: "designer", label: "UI/UX Designer" },
   ];
```
반영완료

FeelsBotMan committed:
refactor: SignUpPage에서 포지션 데이터 구조 변경 및 선택된 포지션 처리 수정

- 포지션 객체의 키를 id에서 positionId로 변경
- 선택된 포지션이 null일 경우 기본값으로 1 설정
- 포지션 버튼의 키와 클릭 핸들러에서 positionId 사용

FeelsBotMan commented on src/pages/SignUpPage.tsx:
```diff
@@ -167,7 +164,12 @@ function SignUpPage() {
-    signUp(data).then(() => {
+    const signUpData = {
+      ...data,
+      positionId: selectedPosition as number,
```
반영완료

FeelsBotMan committed:
fix: SignUpPage에서 선택된 포지션의 기본값을 1로 설정
